### DISCUSSION TITLE
是否有必要在 rc -component 里 以及 antd 里使用 findMaxZindex 的函数, 用于弹窗元素 ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [ant-design/ant-design#43260] [Ideas]
之前遇到的一个自定义 select 的 dropdownRender 里弹窗层级不够的bug, 我想需要一个通用方法, 统一解决 所有这类的 bug

https://github.com/zhangxinxu/quiz/issues/43

**Comment by zombieJ** [comment]
`z-index` 是个复杂的问题，不能简单通过累加来计算。一个例子就是当页面中存在多个 Drawer 在不同的需求下 `z-index` 值累加规则也会不同：
* Drawer 之间没有先后顺序（按照打开顺序累加）
* Drawer 之间有先后顺序（按照节点顺序累加）

而到了 Tooltip Select 这种就会变得更加复杂，比如说 Tour 下，弹出的 Select 到底应该是被 Tour 遮盖还是不遮盖，它属于提示的一部分，还是次要内容。会跟着业务逻辑变化而变化。

所以 antd 中现在是按照组件维度设置默认 `z-index`，而在组件之间则通过打开顺序以原生的 HTML 行为进行排序。

**Comment by vaynevayne** [comment]
Drawer 之间没有先后顺序 和有先后顺序是怎么区分的?  怎么会那么复杂
我理解的是, 后打开的总是最大 zIndex,  毕竟原生的 `<dialog/>` 也是这样的逻辑

**Reply by zombieJ** [reply]
业务场景不同各不相同，例如顶级配置 Drawer 始终在分项配置前，如果打开顶级后关闭再打开分项，再在分项里唤起顶级，顶级配置仍然应该覆盖分项配置。如果后打开始终是最高优先级，那就什么都不用操作，antd 的 Drawer 本来就是利用原生的插入顺序来实现这个效果的。

**Comment by vaynevayne** [comment]
什么 顶级配置 分项配置  听不懂
算了, 
我懂你意思: 不同意

**Reply by zombieJ** [reply]
> 如果后打开始终是最高优先级，那就什么都不用操作，antd 的 Drawer 本来就是利用原生的插入顺序来实现这个效果的。

现在就是利用原生的能力来实现顺序叠加的哈~

> 什么 顶级配置 分项配置 听不懂

做个例子应该更好理解一些：

https://codesandbox.io/s/ji-chu-chou-ti-antd-5-6-2-forked-sm4z2c?file=/demo.tsx

**Comment by vaynevayne** [comment]
<img width="583" alt="image" src="https://github.com/ant-design/ant-design/assets/124666577/80cac618-01a4-49b8-9bdc-567142366560">

怎么感觉有 bug, 第一次link 按钮, 会打开另一个 dialog, 但是刷新以后, 再次 点击link 按钮, 就不打开 dialog了

另外还是没 get 到你的意思, 
`后打开的 zIndex 总是在最上面` 这不是很符合直觉吗, 你举个后打开的不能在最前面的反例子?

**Reply by zombieJ** [reply]
这个就是我说的业务逻辑是和 zIndex 有影响哈~

示例中 知识库配置总是应该在最前面，和它添加顺序无关。这个示例就是为了模拟出如果按照加载顺序展示会出现 bug 的场景。

**Reply by vaynevayne** [reply]
这是需求? 我觉得是 bug
